nucypher-core/src/dkg.rs

```diff
@@ -0,0 +1,115 @@
+use alloc::boxed::Box;
+use alloc::string::String;
+
+use serde::{Deserialize, Serialize};
+
+use crate::conditions::{Conditions, Context};
+
+use crate::versioning::{
+    messagepack_deserialize, messagepack_serialize, ProtocolObject, ProtocolObjectInner,
+};
+
+/// The ferveo variant to use for the decryption share derivation.
+#[derive(PartialEq, Eq, Debug, Serialize, Deserialize, Clone)]
+pub enum FerveoVariant {
```

Comment:
This can just implement `Copy`, so you don't have to clone it explicitly

Reply:
How does this change the usage, do I still need to call `.clone()`?